### DISCUSSION TITLE
Does this work with Cloudflare Workers?

### DISCUSSION THREAD

**Discussion by skyfallwastaken** [vercel/satori#159] [Q&A] [answered]
Hi, I'm thinking of using this package on Cloudflare Workers, does it work on there?

**Comment by shuding** [comment, marked as answer]
Yes it should work.

**Reply by skyfallwastaken** [reply]
Thanks!

**Comment by 0xcaff** [comment]
I tried it and it doesn't seem to work. Run into two problems.

1. eval is used by nbind (https://github.com/charto/nbind/issues/128
) in yoga-layout-prebuilt (related discussion https://github.com/diegomura/react-pdf/issues/510). eval is disallowed in cloudflare workers https://developers.cloudflare.com/workers/runtime-apis/web-standards/#javascript-standards

2. Something causes an error on import.

```
✘ [ERROR] Error on remote worker: ParseError: A request to the Cloudflare API (/accounts/86fdf8ef811b4d0aa8954fa036684ea3/workers/scripts/nouns-agora-dev/edge-preview) failed.

      at throwFetchError
  (/Users/martin/projects/nouns-agora/node_modules/wrangler/wrangler-dist/cli.js:143459:17)
      at fetchResult
  (/Users/martin/projects/nouns-agora/node_modules/wrangler/wrangler-dist/cli.js:143428:5)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async createPreviewToken
  (/Users/martin/projects/nouns-agora/node_modules/wrangler/wrangler-dist/cli.js:143741:29)
      at async createWorkerPreview
  (/Users/martin/projects/nouns-agora/node_modules/wrangler/wrangler-dist/cli.js:143761:17)
      at async start
  (/Users/martin/projects/nouns-agora/node_modules/wrangler/wrangler-dist/cli.js:144411:34) {
    text: 'A request to the Cloudflare API
  (/accounts/86fdf8ef811b4d0aa8954fa036684ea3/workers/scripts/nouns-agora-dev/edge-preview)
  failed.',
    notes: [
      {
        text: 'Uncaught RangeError: Array buffer allocation failed\n' +
          '  at worker.js:12717:20\n' +
          '  at worker.js:39401:5 in
  ../../node_modules/yoga-layout-prebuilt/yoga-layout/dist/entry-browser.js\n' +
          '  at worker.js:85:53 in __require22\n' +
          '  at worker.js:88657:18\n' +
          ' [code: 10021]'
      }
    ],
    location: undefined,
    kind: 'error',
    code: 10021
  }
```

This seems kinda hard to fix from nbind as it looks like nbind has a bunch of native code which generates javascript which is eval'd to create executable bindings. One way around this might be to use something like stretch instead and write our own bindings https://github.com/vislyhq/stretch but that's an entire new solution with unbounded complexity.

**Reply by 0xcaff** [reply]
Actually this works with `satori/wasm` described here https://github.com/vercel/satori#runtime-and-wasm

resvg + satori does cost quite a bit of bundle size. workers has a 1mb compressed default limit https://developers.cloudflare.com/workers/platform/limits/#worker-size

<img width="1840" alt="image" src="https://user-images.githubusercontent.com/3277097/196224050-52640f03-801d-40bc-8dfe-f9811d39d81d.png">

**Reply by shuding** [reply]
Yeah `satori` uses native dependencies so you have to use the WASM version in workers. Regarding the size limit, I think Resvg + Satori ~= 500kb **gzipped**, which is fine for workers.

**Reply by inxilpro** [reply]
@0xcaff any chance you could share more info on how you got this working on Cloudlare?

**Reply by superhighfives** [reply]
I struggled with the same issue, and ended up writing a Vite plugin to get things working on Vite, Remix, and Cloudflare: https://github.com/superhighfives/vite-plugin-wasm-module-workers/ cc @0xcaff
